File: python3/easy/0007.py
class Solution:
    def reverse(self, x):
        INT32_MAX = 2 ** 31 - 1
        INT32_MIN = -2 ** 31
        INT32_MAX_DIV_10 = INT32_MAX / 10
        
        sign = 1 if x >= 0 else -1
        x = abs(x)

        result = 0

        while x:
            x, digit = divmod(x, 10)
            if result > INT32_MAX_DIV_10 or (result == INT32_MAX_DIV_10 and digit > 7):
                return 0
            result = result * 10 + digit

        return result * sign


solution = Solution()

print(solution.reverse(12345))

File: javascript/easy/0007.js
const reverse = x => {
  const INT32_MAX = Math.pow(2, 31) -1;
  const INT32_MAX_DIV_10 = INT32_MAX / 10

  const sign = x < 0 ? -1 : 1;

  let result = 0;
  let abs_result = Math.abs(x)

  while (abs_result > 0) {
    if (result > INT32_MAX || (result == INT32_MAX_DIV_10 && remainder > 7)){
        return 0
    }
     
    result = (result * 10) + (abs_result % 10);
    abs_result = Math.floor(abs_result / 10);
  };

  return result * sign
};

reverse(123);
